core/controllers/contributor_dashboard.py

```diff
@@ -524,3 +524,46 @@ def get(self):
             'topic_names': topic_names
         }
         self.render_json(self.values)
+
+
+class TranslationPreferenceHandler(base.BaseHandler):
+    """Provides the preferred translation language in the
+    contributor dashboard page.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'POST': {
+            'language_code': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_supported_audio_language_code'
+                    }]
+                }
+            }
+        }
+    }
+
+    @acl_decorators.open_access
```

Comment:
Why is this open access? Shouldn't this be limited to logged-in users?

Reply:
Yes, it should be limited to the logged-in users but the "contributor dashboard page" can be accessed by anyone and anyone can select a language to see the opportunities, and whenever one selects, a post request will be sent to save the preference. Now, I am not sure if we should use `@acl_decorators.can_manage_own_account` or `@acl_decorators.open_access`.
Also, If I use `@acl_decorators.can_manage_own_account` then it may lead to errors because when guest users try to change the language, the post request will throw an error. I am facing this particular error in the e2e tests as well.

Reply:
@sagangwee @vojtechjelinek  Is there any way to distinguish between a logged-in user and a guest user in the front-end itself? 
I will then add an if-else statement there itself so that the post request will not be sent if the user is not logged-in to avoid the error.

Reply:
I don't know offhand, but if you look at the logged-in and logged-out views, the CD is showing different things, right?

So maybe look into how it's doing that, and you might be able to use that info as inspiration.

It does seem to me that this handler should assume the user is logged in (and the backend tests should be updated accordingly).

Reply:
Thanks for this tip. This was really helpful.  
I have now changed this to `@acl_decorators.can_manage_own_account` and added an if-else statement in the front-end to make the post call for saving user preference only when the user is logged in by using `UserService`.

Reply:
It still looks like open_access in my Files Changed view.

Reply:
I think now, these changes would be visible to you. I have made another commit for the same.

(PS: I don't know why this commit is not updated on Github. It ain't reflecting at my end too so I made another commit.)